### PR TITLE
feat: add full infra nginx support and opa envoy/ opala configuration (MAPCO-3853)

### PR DESCRIPTION
**Pull request by RonitKissis**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

Related issues: #XXX , #XXX ...
Closes #XXX ...

Further  information:

### CODE REVIEW

File: helm/values-prod.yaml [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+  imagePullSecretName: acr-registry
+  flavor: openshift
+
+authentication:
+  ssl:
+    enabled: false
+    useExternal: false
+    externalSecretName: ""
+  opa:
+    enabled: false
```
Comment (asafMasa) [inline helm/values-prod.yaml:33]: true

File: helm/values-prod.yaml [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+  tag: "v1.5.0"
+
+targetPort: 8080
+internalServicePort: 80
+servicePort: 8080 
+nodePort: 30018 
+
+serviceUrls:
+  exporterTrigger: "http://raster-prod-exporter-trigger-service"
+  jobManager: "http://common-job-manager"
```
Comment (asafMasa) [inline helm/values-prod.yaml:51]: is it for prod?

File: helm/values-prod.yaml [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+jobManagement:
+    exporter:
+      jobType: tilesExport
```
Comment (asafMasa) [inline helm/values-prod.yaml:54]: check how it is defined in raster Prod deployment

File: helm/values-prod.yaml [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+
+env:
+  protocol: TCP
+  logLevel: "warn"
```
Comment (asafMasa) [inline helm/values-prod.yaml:58]: info

File: helm/values-prod.yaml [hunk 5]
```diff
@@ -0,0 +1,246 @@
+global:
+  cloudProvider:
+    dockerRegistryUrl: acrarolibotnonprod.azurecr.io
+    imagePullSecretName: acr-registry
+    flavor: openshift
+  tracing: {}
+  metrics: {}
+  environment: {}
+
+  opalaEnabled: false
```
Comment (asafMasa) [inline helm/values-prod.yaml:10]: true

File: helm/values.yaml [hunk 6]
```diff
@@ -1,54 +1,72 @@
...
   environment: {}
 
+  opalaEnabled: false  
+  authentication:
+    jwtPublicKeySecretName: map-colonies-jwks
+
 enabled: true
-environment: development
-replicaCount: 1
+environment: production
```
Comment (asafMasa) [inline helm/values.yaml:15]: revert to "development"

File: helm/values.yaml [hunk 7]
```diff
@@ -1,54 +1,72 @@
...
+replicaCount: 2
```
Comment (asafMasa) [inline helm/values.yaml:16]: dev - 1

File: helm/values.yaml [hunk 8]
```diff
@@ -1,54 +1,72 @@
...
 initialDelaySeconds: 60
-nodePort: 30030 #for minikube deployment only
+# nodePort: 30030 #for minikube deployment only
```
Comment (asafMasa) [inline helm/values.yaml:18]: why?

File: helm/values-prod.yaml [hunk 9]
```diff
@@ -0,0 +1,246 @@
+global:
```
Comment (asafMasa) [inline helm/values-prod.yaml:1]: move this file to dedicated deployment repo

File: helm/values.yaml [hunk 10]
```diff
@@ -1,54 +1,72 @@
 global:
-  cloudProvider: {}
+  cloudProvider:
+    dockerRegistryUrl: acrarolibotnonprod.azurecr.io
```
Comment (asafMasa) [inline helm/values.yaml:3]: revert to empty config as it is public repo

File: helm/values.yaml [hunk 11]
```diff
@@ -1,54 +1,72 @@
...
 initialDelaySeconds: 60
-nodePort: 30030 #for minikube deployment only
+# nodePort: 30030 #for minikube deployment only
 resetOnConfigChange: true
 imagePullPolicy: Always
 
 cloudProvider:
-  dockerRegistryUrl: my-registry-url.io
-  imagePullSecretName: 'my-registry-secret'
+  dockerRegistryUrl: acrarolibotnonprod.azurecr.io
```
Comment (asafMasa) [inline helm/values.yaml:23]: revert

File: helm/values.yaml [hunk 12]
```diff
@@ -1,54 +1,72 @@
...
+    customHeaderName: "x-api-key,content-type"
+
 caSecretName: ''
 caPath: '/usr/local/share/ca-certificates'
 caKey: 'ca.crt'
 
 image:
   repository: export-management
-  tag: 'latest'
+  tag: "v1.5.0"
```
Comment (asafMasa) [inline helm/values.yaml:42]: tag: 'latest'

File: helm/values.yaml [hunk 13]
```diff
@@ -1,54 +1,72 @@
...
+
+targetPort: 8080
+internalServicePort: 80
+servicePort: 8080 
+nodePort: 30018 
 
 serviceUrls:
-  exporterTrigger: "http://job-manager-service-url"
-  jobManager: "http://job-manager-service-url"
+  exporterTrigger: "http://raster-prod-exporter-trigger-service"
```
Comment (asafMasa) [inline helm/values.yaml:50]: dev trigger

File: helm/values.yaml [hunk 14]
```diff
@@ -1,54 +1,72 @@
...
+  jobManager: "http://common-job-manager"
```
Comment (asafMasa) [inline helm/values.yaml:51]: dev job-manager

File: helm/values.yaml [hunk 15]
```diff
@@ -1,54 +1,72 @@
...
     exporter:
       jobType: tilesExport
 
 env:
-  port: 8080
-  internalPort: 80
-  targetPort: 8080
   protocol: TCP
-  logLevel: info
+  logLevel: "warn"
```
Comment (asafMasa) [inline helm/values.yaml:58]: info

File: helm/values.yaml [hunk 16]
```diff
@@ -1,54 +1,72 @@
...
   logPrettyPrintEnabled: false
   responseCompressionEnabled: true
   disableHttpClientLogs: false
-  requestPayloadLimit: 1mb
-  serviceWebhookEndpoint: https://export-management-test-integration.apps.j1lk3njp.eastus.aroapp.io/export-tasks/webhook/
+  requestPayloadLimit: 10mb
+  serviceWebhookEndpoint: https://export-management.apps.j1lk3njp.eastus.aroapp.io/export-tasks/webhook/
```
Comment (asafMasa) [inline helm/values.yaml:63]: dev

File: helm/values.yaml [hunk 17]
```diff
@@ -1,54 +1,72 @@
...
   tracing:
     enabled: false
-    url: http://localhost:55681/v1/trace
+    url: http://infra-monitoring-opentelemetry-collector.infra-services.svc.cluster.local:4317
```
Comment (asafMasa) [inline helm/values.yaml:66]: for dev

File: helm/values.yaml [hunk 18]
```diff
@@ -1,54 +1,72 @@
...
   metrics:
     enabled: false
-    url: http://localhost:55681/v1/metrics
+    url: http://infra-monitoring-opentelemetry-collector.infra-services.svc.cluster.local:4317/v1/metrics
```
Comment (asafMasa) [inline helm/values.yaml:69]: for dev

File: helm/values.yaml [hunk 19]
```diff
@@ -64,51 +82,165 @@ resources:
       cpu: 100m
       memory: 128Mi
 
+
 nginx:
   enabled: true
+  fullnameOverride: "export-management-export-management-nginx"
   replicaCount: 2
```
Comment (asafMasa) [inline helm/values.yaml:89]: 1

File: helm/values.yaml [hunk 20]
```diff
@@ -64,51 +82,165 @@ resources:
...
+  nginxTargetPort: &nginxTargetPort 8080
+  targetPort: *nginxTargetPort  # --> if global.opalaEnabled is true use *nginxTargetPort else use *envoyPort | if no authorization is needed use *nginxTargetPort
+  nodePort: 30003
+
   authorization:
     enabled: true
-    domain: 'rasterExport'
-    url: http://opa-la-opa-service/v1/data/http/authz/decision
+    domain: "rasterExport"
+    url: http://opa-int-opa-service.infra-services/v1/data/http/authz/decision
```
Comment (asafMasa) [inline helm/values.yaml:137]: for dev

File: helm/templates/_helpers.tpl [hunk 21]
```diff
@@ -128,3 +128,18 @@ Returns the tracing url from global if exists or from the chart's values
     {{- .Values.env.metrics.url -}}
 {{- end -}}
 {{- end -}}
+
+
+{{/*
+Create export-management nginx configmap name as used by the service name label.
```
Comment (asafMasa) [inline helm/templates/_helpers.tpl:134]: please create "_resources.tpl" as is done in raster services

File: helm/values.yaml [hunk 22]
```diff
@@ -1,54 +1,72 @@
...
-nodePort: 30030 #for minikube deployment only
+# nodePort: 30030 #for minikube deployment only
 resetOnConfigChange: true
 imagePullPolicy: Always
 
 cloudProvider:
-  dockerRegistryUrl: my-registry-url.io
-  imagePullSecretName: 'my-registry-secret'
+  dockerRegistryUrl: acrarolibotnonprod.azurecr.io
+  imagePullSecretName: acr-registry
```
Comment (asafMasa) [inline helm/values.yaml:24]: revert

**Review by asafMasa — CHANGES_REQUESTED** [review]
Looking good

File: helm/values.yaml (hunk 18 above)
Comment (RonitKissis) [reply helm/values.yaml:69]: this is the url roee told me to use

File: helm/values.yaml (hunk 20 above)
Comment (RonitKissis) [reply helm/values.yaml:137]: this is the url for the opala service for int/prod

File: helm/values-prod.yaml (hunk 1 above)
Comment (RonitKissis) [reply helm/values-prod.yaml:33]: removed

File: helm/values-prod.yaml (hunk 3 above)
Comment (RonitKissis) [reply helm/values-prod.yaml:54]: as written

File: helm/values.yaml (hunk 8 above)
Comment (RonitKissis) [reply helm/values.yaml:18]: located elsewhere

File: helm/values.yaml [hunk 23]
```diff
@@ -1,72 +1,68 @@
...
-    imagePullSecretName: acr-registry
+    dockerRegistryUrl: my-registry-url.io
+    imagePullSecretName: 'my-registry-secret'
     flavor: openshift
   tracing: {}
   metrics: {}
   environment: {}
 
-  opalaEnabled: false  
+  opalaEnabled: true
```
Comment (asafMasa) [inline helm/values.yaml]: false

**Review by asafMasa — APPROVED** [review]

**Pull request merged**